### PR TITLE
feat: define allowed types to be published

### PR DESCRIPTION
**Pull request by ReidyT**
ref of graasp/graasp#1104

### CODE REVIEW

**Review by spaenleh — CHANGES_REQUESTED** [review]
Nice changes. 

I would suggest maybe to group related elements in a single "namespace", like an object that exposes multiple methods. 

I am also not sure if you need to export the array of allowed item types, as well as a positive and a negative function. What do you think ?

File: src/item/itemType.ts [hunk 1]
```diff
@@ -12,3 +12,4 @@ export const ItemType = {
   ETHERPAD: 'etherpad',
 } as const;
 export type ItemTypeUnion = UnionOfConst<typeof ItemType>;
+export type ValuesOfItemType = (typeof ItemType)[keyof typeof ItemType];
```
[the remark this answers is not in the document]
Comment (ReidyT) [reply src/item/itemType.ts:15]: No, I will use `ItemTypeUnion`, I didn't check the other types 😓

File: src/itemPublished/itemPublished.ts [hunk 2]
```diff
@@ -16,3 +22,9 @@ export interface PackedItemPublished {
   item: PackedItem;
   totalViews: number;
 }
+
+export const ALLOWED_PUBLISHED_TYPES: ValuesOfItemType[] = [ItemType.FOLDER];
+export const itemTypeAllowedToBePublished = (itemType: ValuesOfItemType) =>
+  ALLOWED_PUBLISHED_TYPES.includes(itemType);
+export const itemTypeNotAllowedToBePublished = (ItemType: ValuesOfItemType) =>
+  !itemTypeAllowedToBePublished(ItemType);
```
[the remark this answers is not in the document]
Comment (ReidyT) [reply src/itemPublished/itemPublished.ts:30]: No I simplified to only one function, thanks for this comment !

File: src/itemPublished/itemPublished.ts [hunk 3]
```diff
@@ -16,3 +22,17 @@ export interface PackedItemPublished {
...
   totalViews: number;
 }
+
+export class PublishableItemTypeChecker {
+  private static readonly ALLOWED_PUBLISHED_TYPES: ItemTypeUnion[] = [
+    ItemType.FOLDER,
+  ];
+
+  public static isItemTypeAllowedToBePublished(itemType: ItemTypeUnion) {
+    this.ALLOWED_PUBLISHED_TYPES.includes(itemType);
```
Comment (spaenleh) [inline src/itemPublished/itemPublished.ts:32]: Are you missing the return here ? I don't think JS has implicit returns (yet).

**Review by spaenleh — CHANGES_REQUESTED** [review]
Nice change ! I think you just have missing return 😉

**Review by Defferrard — APPROVED** [review]
LGTM !

File: src/itemPublished/itemPublished.ts (hunk 3 above)
Comment (ReidyT) [reply src/itemPublished/itemPublished.ts]: Yes I saw a little bit too late ^^

**Review by spaenleh — APPROVED** [review]
Nice ! 🌴

**Pull request merged**